packages/kbn-cell-actions/src/actions/utils.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { KBN_FIELD_TYPES } from '@kbn/field-types';
+import { isBoolean, isNumber, isString } from 'lodash/fp';
+import { Serializable, SerializableArray } from '@kbn/utility-types/src/serializable';
+import { DefaultActionsSupportedValue, NonNullableSerializable } from './types';
+import { CellActionFieldValue } from '../types';
+
+export const SUPPORTED_KBN_TYPES = [
+  KBN_FIELD_TYPES.DATE,
+  KBN_FIELD_TYPES.IP,
+  KBN_FIELD_TYPES.STRING,
+  KBN_FIELD_TYPES.NUMBER,
+  KBN_FIELD_TYPES.BOOLEAN,
+];
+
+export const isTypeSupportedByDefaultActions = (kbnFieldType: KBN_FIELD_TYPES) =>
+  SUPPORTED_KBN_TYPES.includes(kbnFieldType);
+
+const isNonNullablePrimitiveValue = (
+  value: CellActionFieldValue
+): value is string | number | boolean => isString(value) || isNumber(value) || isBoolean(value);
+
+const isNonMixedTypeArray = (
+  value: Array<string | number | boolean>
+): value is string[] | number[] | boolean[] => value.every((v) => typeof v === typeof value[0]);
+
+export const isValueSupportedByDefaultActions = (
+  value: NonNullableSerializable[]
+): value is DefaultActionsSupportedValue =>
+  value.every(isNonNullablePrimitiveValue) && isNonMixedTypeArray(value);
```

Comment:
Is the `value.every(isNonNullablePrimitiveValue)` needed? The value parameter is already `NonNullableSerializable[]`?

Reply:
`isNonNullablePrimitiveValue` is required to drop non-primitive values like `SerializableArray` and `SerializableRecord`.
The confusion is due to the bad naming. `isNonNullablePrimitiveValue` checks if the value is `string`, `number`, or `boolean`, which are primitive and not nullable. I will inline the function to avoid naming it. 😆